### PR TITLE
Simantov

### PR DESCRIPTION
**Pull request by jamberusimantov**

### CODE REVIEW

File: client/src/components/layout/index.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import "./style.css";
+import AdminRouter from '../../AdminRouter'
+import {Link} from 'react-router-dom';
+
+
+
+import { Layout, Menu, Breadcrumb } from "antd";
+import "antd/dist/antd.css";
+
+// import {
```
Comment (yonatanpa) [inline client/src/components/layout/index.tsx:11]: remove comments

File: client/src/components/layout/index.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+          <Content style={{ padding: "0 24px", minHeight: 280 }}>
+            
+
+                          <AdminRouter/>
+
+
+          </Content>
+        </Layout>
+      </Content>
+      {/* <Footer style={{ textAlign: "center" }}>
```
Comment (yonatanpa) [inline client/src/components/layout/index.tsx:51]: remove comments

File: client/src/components/layout/style.css [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    background: #fff;
+  }
+
+.ant-layout ant-layout-rtl{
+  width: 100%;
+  height: 100vh;
+}
+
+
+  /* #components-layout-demo-top-side .logo {
```
Comment (yonatanpa) [inline client/src/components/layout/style.css:29]: remove comments

File: client/src/utils/login.utils.ts [hunk 4]
```diff
@@ -61,55 +60,56 @@ export default new LogIn();
 
 
 
-  //1. post User to db by admin
-  // registerUser({ company: 'google', email: 'jamber@google.com' }, 'hr')
-  // .then(data => {console.log(data.data.link)}); 
+//   //1. post User to db by admin
```
Comment (yonatanpa) [inline client/src/utils/login.utils.ts:63]: is this code in use? if not, remove it
also, why is it commented out twice?

File: client/src/utils/login.utils.ts [hunk 5]
```diff
@@ -1,6 +1,5 @@
 import { API } from '../app.utils';
 import Http from './http.utils';
-
 class LogIn {
     token = 'token';
     logList: any[] = [];
```
Comment (yonatanpa) [inline client/src/utils/login.utils.ts:5]: can we use a better type here? (not any)

File: server/api/register/register_ctrl.js [hunk 6]
```diff
@@ -49,26 +49,35 @@ const signToken = (req, res, payload, message, emailVerification = false) => {
             if (err) throw new Error(`error on sign token ${err}`)
             const dataToUpdate = !emailVerification ? { _id, isActive: true, token: `Bearer ${token}` } : { _id, token: `Bearer ${token}` }
             const updateDocSuccessCb = async(data) => {
-                const link = new URL(`${baseApi}/signUp/${role}/${token}`)
+                const client = process.env.NODE_ENV === 'production' ?
```
Comment (yonatanpa) [inline server/api/register/register_ctrl.js:52]: move the urls to a configuration file (json or .env)

Comment (yonatanpa) [reply server/api/register/register_ctrl.js:52]: you can use dotenv that is already in the project for this

File: server/email/email.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    try {
+        const transporter = nodemailer.createTransport({
+            service: 'gmail',
+            auth: {
+                user: process.env.USER,
+                pass: process.env.PASS
+            }
+        });
+        return await transporter.sendMail(email);
+    } catch (error) { return { success: false, error } } finally {}
```
Comment (yonatanpa) [inline server/email/email.js:57]: remove empty finally block

File: server/email/email.js [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+    const email = createMail(recipientAddress, link)
+    if (typeof email === 'string') throw new Error(email);
+
+    const { errors, isValid } = mailValidation(email)
+    if (!isValid) throw errors;
+    try {
+        const transporter = nodemailer.createTransport({
+            service: 'gmail',
+            auth: {
+                user: process.env.USER,
```
Comment (yonatanpa) [inline server/email/email.js:52]: maybe rename the variables to GMAIL_USER GMAIL_PASSWORD to make it clear that these credentials are for email

**Review by liori142 — APPROVED** [review]
nice and clean

**Review by yonatanpa — CHANGES_REQUESTED** [review]
Nice work:)
Please check out my comments and let me know if something isn't clear

**Pull request merged**